File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(index, visited, computers):
    visited[index] = 1
    for idx, com in enumerate(computers[index]):
        if idx == index or com == 0 or visited[idx]:
            continue
        dfs(idx, visited, computers)
        
    

def solution(n, computers):
    answer = 0
    visited = [0 for i in range(n)]
    for i in range(n):
        if visited[i] == 1:
            continue
        dfs(i, visited, computers)
        answer += 1
        
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0;
    const visited = [];

    function DFS(index, visited) {
        visited.push(index);
        const targets = [];
        for (let i = 0; i < n; i++) {
            if (i !== index && computers[index][i] === 1) {
                targets.push(i);
            }
        }
        for (let i = 0; i < targets.length; i++) {
            if(!visited.includes(targets[i])) {
                DFS(targets[i], visited);
            }
        }
    }
    
    for (let i = 0; i < n; i++) {
        if(!visited.includes(i)) {
            DFS(i, visited);
            answer += 1;
        }
    }
    return answer;
}